mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt

```diff
@@ -57,16 +70,24 @@ class CustomInputTypePreference constructor(context: Context, attrs: AttributeSe
                 }
             }
             super.onBindDialogView(view)
+            if (arguments?.getBoolean(KEY_IS_PREF_SECRET) == true) {
+                editor?.setText(context?.getSecretPrefs()?.getString(this.preference.key))
```

Comment:
Where/how is the value written to the secret prefs?

I think a better approach would be writing a (trivial) `PreferenceDataStore` around the secret prefs and attaching that to the respective prefs from within `PreferenceActivity`.

Reply:
Implementation to above suggestion: https://github.com/maniac103/openhab.android/commits/backup ... IMHO it simplifies logic quite a bit.

Reply:
Indeed, that's better.